Mathlib/LinearAlgebra/Matrix/Spectrum.lean

```diff
@@ -34,6 +34,23 @@ namespace IsHermitian
 section DecidableEq
 
 variable [DecidableEq n]
+
+section RankMulUnitaryTheorems
+
+variable {m R : Type*} [CommRing R] [StarRing R]
+
+@[simp]
+theorem rank_unitary_mul (A : unitaryGroup n R) (B : Matrix m n R) :
+    rank (B * (A : Matrix n n R)) = rank B :=
+  rank_mul_units (unitary.toUnits A) _
+
+@[simp]
+theorem rank_mul_unitary [CommRing R] [Fintype m] [StarRing R] (A : unitaryGroup n R)
```

Comment:
As written, this runs the risk of having *duplicate* `[CommRing R]` and `[StarRing R]` hypotheses in the statement, since these were already declared with `variable`.
```suggestion
theorem rank_mul_unitary [Fintype m] (A : unitaryGroup n R)
```